src/main/java/com/indeed/jiraactions/api/ApiCaller.java

```diff
@@ -3,173 +3,94 @@
-    private String upstream = null;
-    private String cookies = "";
-    private String pinnedNode = null;
 
     public ApiCaller(final JiraActionsIndexBuilderConfig config) {
         this.config = config;
         this.authentication = getBasicAuth();
+        this.client = new OkHttpClient.Builder()
+                .connectionPool(new ConnectionPool())
+                .connectTimeout(2, TimeUnit.MINUTES)
+                .readTimeout(20000, TimeUnit.MILLISECONDS)
+                .build();
     }
 
     public JsonNode getJsonNode(final String url) throws IOException {
-        HttpsURLConnection urlConnection = null;
-        Map<String, List<String>> headers = null;
-        BufferedReader br = null;
-        String apiResults = null;
-        try {
-            urlConnection = getURLConnection(url);
-            headers = urlConnection.getRequestProperties();
-            final InputStream in = urlConnection.getInputStream();
-            br = new BufferedReader(new InputStreamReader(in));
-            apiResults = br.readLine();
-
-            final String anodeId = urlConnection.getHeaderField("X-ANODEID");
-
-            if(!Objects.equals(pinnedNode, anodeId)) {
-                if(pinnedNode != null) {
-                    log.warn("Expected X-ANODEID={} but found {}", pinnedNode, anodeId);
-                }
-                final Map<String, List<String>> responseHeaders = urlConnection.getHeaderFields();
-                final List<String> cookies = responseHeaders.get("Set-Cookie");
-                if(cookies != null) {
-                    for (final String cookie : cookies) {
-                        if (cookie.startsWith("JSESSIONID=")) {
-                            final int start = "JSESSIONID=".length();
-                            final int end = cookie.contains(";") ? cookie.indexOf(";") : cookie.length();
-                            jsessionId = cookie.substring(start, end);
-                        } else if (cookie.startsWith("upstream")) {
-                            final int start = "upstream=".length();
-                            final int end = cookie.contains(";") ? cookie.indexOf(";") : cookie.length();
-                            upstream = cookie.substring(start, end);
-                        }
-                    }
-                    if (jsessionId != null || upstream != null) {
-                        setCookies();
-                        pinnedNode = anodeId;
-                        log.info("Set JSESSION={};upstream={}. Pinning to X-ANODEID={}",
-                                jsessionId, upstream, anodeId);
-                    }
-                }
-            }
-            return objectMapper.readTree(apiResults);
-        } catch (final IOException e) {
-            final StringBuilder sb = new StringBuilder();
-
-            sb.append("{");
-            sb.append("\"Request\": {");
+        final OkHttpClient newClient = client.newBuilder().build();
```

Comment:
Why are we building a new client here? Why not just use the existing one instantiated in the constructor?